sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeMapData.java

```diff
@@ -143,7 +143,7 @@ public void writeExternal(ObjectOutput out) throws IOException {
   }
 
   @Override
-  public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+  public void readExternal(ObjectInput in) throws IOException {
```

Comment:
I think this OK but I have this recollection in the back of my mind that it's a special serialization method that needs this particular signature. I think I'm thinking of the `Serializable` mechanism's requirements. Probably fine to remove this.